Criterion/Internal.hs

```diff
@@ -111,33 +112,45 @@ runAndAnalyse :: (String -> Bool) -- ^ A predicate that chooses
               -> Benchmark
               -> Criterion ()
 runAndAnalyse select bs = do
-  mbRawFile <- asks rawDataFile
-  (rawFile, handle) <- liftIO $
-    case mbRawFile of
+  mbJsonFile <- asks jsonFile
+  (jsonFile, handle) <- liftIO $
+    case mbJsonFile of
       Nothing -> do
         tmpDir <- getTemporaryDirectory
-        openBinaryTempFile tmpDir "criterion.dat"
```

Comment:
Oops!  Good catch.  Specifically, "criterion.dat" should only be used  when `rawDataFile` is *not* set.

But you are right that we forgot to restore writing of binary reports when `rawDataFile` *is* set.  I.e. just below where it calls `report rpts; json rpts; junit rpts`.